service/tbc/rpc_test.go

```diff
@@ -56,42 +55,31 @@ func TestBlockHeadersByHeightRaw(t *testing.T) {
 		conn: protocol.NewWSConn(c),
 	}
 
-	var lastErr error
 	var response tbcapi.BlockHeadersByHeightRawResponse
-	for {
-		select {
-		case <-time.After(1 * time.Second):
-		case <-ctx.Done():
-			t.Fatal(ctx.Err())
-		}
-		lastErr = nil
-		err = tbcapi.Write(ctx, tws.conn, "someid", tbcapi.BlockHeadersByHeightRawRequest{
-			Height: 55,
-		})
-		if err != nil {
-			lastErr = err
-			continue
-		}
-
-		var v protocol.Message
-		err = wsjson.Read(ctx, c, &v)
-		if err != nil {
-			lastErr = err
-			continue
-		}
+	select {
+	case <-time.After(1 * time.Second):
+	case <-ctx.Done():
+		t.Fatal(ctx.Err())
+	}
+	err = tbcapi.Write(ctx, tws.conn, "someid", tbcapi.BlockHeadersByHeightRawRequest{
+		Height: 55,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		if v.Header.Command == tbcapi.CmdBlockHeadersByHeightRawResponse {
-			if err := json.Unmarshal(v.Payload, &response); err != nil {
-				t.Fatal(err)
-			}
-			break
-		} else {
-			lastErr = fmt.Errorf("received unexpected command: %s", v.Header.Command)
-		}
+	var v protocol.Message
+	err = wsjson.Read(ctx, c, &v)
+	if err != nil {
```

Comment:
I think this err assignment could be nested (and in other functions below):
```suggestion
	if err = wsjson.Read(ctx, c, &v); err != nil {
```